packages/story-editor/src/components/canvas/useInsertElement.js

```diff
@@ -71,7 +71,11 @@ function useInsertElement() {
       }
 
       try {
-        const imageSrcProxied = getProxiedUrl(resource, imageSrc);
+        const needsProxy =
+          type === 'image'
+            ? resource?.needsProxy
+            : await checkResourceAccess(imageSrc);
+        const imageSrcProxied = getProxiedUrl({ needsProxy }, imageSrc);
```

Comment:
Why not just

```js
const imageSrcProxied =
          type === 'video' ? imageSrc : getProxiedUrl(resource, imageSrc);
```

Reply:
I don't see the need for `checkResourceAccess` when we already have `resource.needsProxy`

Reply:
What if you are filtering in ( like we know some third parties are doing ) from a different media source? The fact is, poster and video could be served from different domains. It is wise here to be careful and just call `checkResourceAccess` for posters, as we do not know if they need a proxy or not. 

> const imageSrcProxied = type === 'video' ? imageSrc : getProxiedUrl(resource, imageSrc);

This feels extremely brittle.